proto/rill/runtime/v1/resources.proto

```diff
@@ -396,9 +396,10 @@ message Chart {
 
 message ChartSpec {
   string title = 1;
-  string query_name = 2;
-  string query_args_json = 3;
-  string vega_lite_spec = 4;
+  string metrics_sql = 2;
+  string api = 3;
+  map<string, string> args = 4;
```

Comment:
Instead of putting the YAML properties here, I would suggest having `data_resolver` and `data_resolver_properties` mirroring the `APISpec`. And then we should have one common YAML structure for parsing a `data:` clause (which we will want to use in all of APIs, charts, alerts, reports, etc.).

This would be similar to the `parseScheduleYAML` function (see [here](https://github.com/rilldata/rill/blob/9b768c083048a1897955b80b6bacb5e002feb1c3/runtime/compilers/rillv1/parse_source.go#L107-L108)) in the compiler. We might have this:

```go
type DataYAML struct {
  SQL string `...`
  MetricsSQL string `...`
  // Can add others. Like "api" for referencing another existing API.
}

func parseDataYAML(raw *DataYAML) (resolver string, resolverProps map[string]any, error) {
  ...
}

// In charts:
type ChartYaml struct {
  // ...
  Data *DataYAML `yaml:"data"`
}

// In alerts:
type AlertYAML struct {
  // ...
  Data *DataYAML `yaml:"data"` // This will eventually replace/deprecate the "query:" property
}

// In APIs:
type APIYAML struct {
  // ...
  Data *DataYAML `yaml:",inline"` // Inline since API's don't nest under "data":
}

// etc.
```